src/main/java/FlightsReporter.groovy

```diff
@@ -0,0 +1,257 @@
+    }
+
+    static boolean isThereAFlightRunningLate(List<Flight> flights) {
+        flights.any {it.runningLate }
+    }
+
+    static List<Flight> getFlightsRunningLate(List<Flight> flights) {
+        flights.findAll{it.runningLate }
+    }
+
+    static List<Flight> getAllFlightsDepartingAfter(List<Flight> flights, LocalDateTime date) {
+        flights.findAll {it.departure >= date }
+    }
+
+    static List<Flight> getAllFlightsDepartingBefore(List<Flight> flights, LocalDateTime date) {
+        flights.findAll{ it.departure < date }
+    }
+//
+    static List<Flight> getAllFlightsArrivingBefore(List<Flight> flights, LocalDateTime date) {
+        flights.findAll {it.arrival < date }
+    }
+
+    static List<Flight> getAllFlightsArrivingAfter(List<Flight> flights, LocalDateTime date) {
+        flights.findAll {it.arrival >= date }
+    }
+
+    static List<Flight> getAllFlightsShorterThan(List<Flight> flights, int amountInMinutes) {
+        flights.findAll() {flightLengthInMinutes(it) < amountInMinutes }
+    }
+
+    static List<Flight> getAllFlightsLongerThan(List<Flight> flights, int amountInMinutes) {
+        flights.findAll { flightLengthInMinutes(it) > amountInMinutes}
+    }
+
+    static List<Flight> getAllFlightWithDuration(List<Flight> flights, int duration) {
+        flights.findAll {flightLengthInMinutes(it) == duration }
+    }
+
+    static List<Flight> getAllFlightsWithPassengersAbove(List<Flight> flights, int passengerAmount) {
+        flights.findAll{it.passengers >= passengerAmount }
+    }
+
+    static List<Flight> getAllFlightsWithPassengersBelow(List<Flight> flights, int passengerAmount) {
+        flights.findAll { it.passengers < passengerAmount }
+    }
+
+    static Flight getFlightWithMostPassengers(List<Flight> flights) {
+        flights.max {it.passengers}
+    }
+
+    static Flight getFlightWithLeastPassengers(List<Flight> flights) {
+        flights.min {it.passengers}
+    }
+
+    static void changeDestinationTo(List<Flight> flights, Airports from, Airports to) {
+        flights.each {
+            if (it.destination == from)
+                it.destination = to
+        }
+    }
+
+    static void changeStartingAirport(List<Flight> flights, Airports from, Airports to) {
+        flights.each {
+            if (it.from == from)
+                it.from == to
+        }
+    }
+
+    static void changeDepartureDate(List<Flight> flights, String flightID, LocalDateTime newDate) {
+        flights.find {it.id == flightID }.departure = newDate
+    }
+
+    static void changeArrivalDate(List<Flight> flights, String flightID, LocalDateTime newDate) {
+        flights.find {it.id = flightID }.arrival = newDate
+    }
+
+    static void setRunningLateToAirport(List<Flight> flights, Airports destination) {
+        flights.each {
+            if (it.destination == destination)
+                it.runningLate = true
+        }
+    }
+
+
+
+    static void main(String[] args) {
```

Comment:
If you have created unit tests with the same data, you can remove the MAIN() method. The tests show how to use your new code and also you can test it that way, so main() is no longer needed for this library